### PR TITLE
feat: [IOBP-636,IOBP-646] Add legacy transactions list

### PR DESCRIPTION
**Pull request by Hantex9**
## Short description
This PR implements the ability to view “legacy” transactions that are not available through biz-events at the moment. 

## List of changes proposed in this pull request
- Added a legacy list screen that retrieves the old transactions not available with biz-events;
- Connected the legacy list screen with the previous transaction details that shows transaction info with a reduced amount of data;
- Added a CTA to access to this new page from:
   - `PaymentsHomeEmptyScreenContent`
   - Into the biz-events transaction list when the user reaches the end of the list.

## How to test
- Checkout the dev-server with the following PR: https://github.com/pagopa/io-dev-api-server/pull/369
- Open the new payment landing page and try to reach the CTA from the entry point listed above, you should be able to see the previous transactions with less data shown.

## Preview

https://github.com/pagopa/io-app/assets/34343582/7e759311-d0f6-48af-962e-edb28c0bb682

### CODE REVIEW

File: ts/features/payments/transaction/components/PaymentsLegacyAttachmentBottomSheet.tsx [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    component: getModalContent(),
+    title: (
+      <H4>
+        {I18n.t("features.payments.transactions.legacy.bottomSheet.title")}
+      </H4>
+    )
+  });
+
+  return { ...modal };
+};
```
Comment (mastro993) [inline ts/features/payments/transaction/components/PaymentsLegacyAttachmentBottomSheet.tsx:48]: We could simplify this hook as follows.
The `useIOBottomSheet` hook internally uses the `H4` typography component to render the title.
And we can return directly the composed hook, avoiding to use of the spread operator.
What do you think?

```suggestion
/**
 * This custom hook, usePaymentsLegacyAttachmentBottomSheet, is designed to display a bottom sheet
 * containing detailed information about where the user can find the legacy transaction recepit.
 */
const usePaymentsLegacyAttachmentBottomSheet = (): IOBottomSheetModal =>
  useIOBottomSheetAutoresizableModal({
    title: I18n.t("features.payments.transactions.legacy.bottomSheet.title"),
    component: (
      <View>
        <ParagraphWithTitle
          title={I18n.t(
            "features.payments.transactions.legacy.bottomSheet.noticeHeading"
          )}
          body={I18n.t(
            "features.payments.transactions.legacy.bottomSheet.noticeBody"
          )}
        />
        <VSpacer size={24} />
        <ParagraphWithTitle
          title={I18n.t(
            "features.payments.transactions.legacy.bottomSheet.allNoticesHeading"
          )}
          body={I18n.t(
            "features.payments.transactions.legacy.bottomSheet.allNoticesBody"
          )}
        />
        <VSpacer size={48} />
      </View>
    )
  });
```

**Review by mastro993 — CHANGES_REQUESTED** [review]

File: locales/en/index.yml [hunk 2]
```diff
@@ -3068,6 +3068,21 @@ features:
...
+          body: Se stai cercando la ricevuta di un avviso pagoPA che hai pagato in passato, rivolgiti all’ente creditore.
+        banner:
+          content: Stiamo generando le ricevute di pagamento in formato PDF.
+          action: Scopri di più
+        title: Operazioni precedenti
+        bottomSheet:
+          title: Dove trovo la ricevuta di pagamento?
+          noticeHeading: Avvisi pagati di recente
+          noticeBody: "Stiamo aggiornando i nostri sistemi: non appena pronte, troverai le ricevute in formato PDF nella sezione Pagamenti."
+          allNoticesHeading: Tutti gli avvisi
```
Comment (thisisjp) [inline locales/en/index.yml:3084]:
```suggestion
          allNoticesHeading: Tutti gli altri avvisi
```

File: locales/it/index.yml [hunk 3]
```diff
@@ -3068,6 +3068,21 @@ features:
...
+          body: Se stai cercando la ricevuta di un avviso pagoPA che hai pagato in passato, rivolgiti all’ente creditore.
+        banner:
+          content: Stiamo generando le ricevute di pagamento in formato PDF.
+          action: Scopri di più
+        title: Operazioni precedenti
+        bottomSheet:
+          title: Dove trovo la ricevuta di pagamento?
+          noticeHeading: Avvisi pagati di recente
+          noticeBody: "Stiamo aggiornando i nostri sistemi: non appena pronte, troverai le ricevute in formato PDF nella sezione Pagamenti."
+          allNoticesHeading: Tutti gli avvisi
```
Comment (thisisjp) [inline locales/it/index.yml:3084]:
```suggestion
          allNoticesHeading: Tutti gli altri avvisi
```

**Review by thisisjp — CHANGES_REQUESTED** [review]
you forgot a word from the Figma file ;)

**Review by thisisjp — APPROVED** [review]

**Pull request merged**